app/frontend/courses/review/components/CoursesReview__ChecklistShow.res

```diff
@@ -172,143 +173,174 @@ let make = (
         </button>
       </div>
       <div className="border bg-white rounded-lg py-2 md:py-4 mt-2 space-y-4">
-        {Js.Array.mapi((reviewChecklistItem, itemIndex) =>
-          <Spread
-            props={"data-checklist-item": string_of_int(itemIndex)} key={string_of_int(itemIndex)}>
-            <div>
-              <h4 className="relative text-sm font-semibold mt-2 md:mt-0 px-6 w-full md:w-4/5">
-                <div className={checklistItemCheckedClasses(itemIndex, selection)} />
-                {ReviewChecklistItem.title(reviewChecklistItem)->str}
-              </h4>
-              <div className="space-y-3 pt-3"> {Js.Array.mapi((checklistItem, resultIndex) =>
```

Comment:
@harigopal

I think the name of the variable checklistItem should be replaced with resultlistItem as we have result array for each checklistitem.

type t = {
  title: string,
  result: array<CoursesReview__ReviewChecklistResult.t>,
}

what do you think ?

Reply:
@yash-learner I think I'd rename it to `reviewChecklistResults`, given that that's the underlying module-type name: `array<ReviewChecklistResult.t>`.

Looking one layer below, it looks like the naming of the key `result` in the `ReviewChecklistItem` module is incorrect. That should have been a plural - `results`.

Reply:
> @yash-learner I think I'd rename it to `reviewChecklistResults`, given that that's the underlying module-type name: `array<ReviewChecklistResult.t>`.

After spending sometime on code at that context changed variable name to `reviewChecklistResult` as only single result will be going to the map item 

[Even in doc it was called with singular name when an array of elements are used](https://rescript-lang.org/docs/manual/latest/api/js/array-2#mapi:~:text=let%20product%20%3D%20(item%2C%20index)%20%3D%3E%20item%20*%20index) and even for the outer loop it called with singular name `reviewChecklistItem`
 
> Looking one layer below, it looks like the naming of the key `result` in the `ReviewChecklistItem` module is incorrect. That should have been a plural - `results`.

Changed to `results` in frontend related and key name in type

Even the backend `graphql` `review_checklist_type.rb` file also have field name has `result`. Did not change to `results` because it might lead to data migration. Is changing this required and is it necessary?

----

I see at some places in the codebase we have used `Js.Array.map` but do not see a map for array in rescript documentation and able to see only for `Js.Array2`

https://rescript-lang.org/docs/manual/latest/api/belt/array#sidebar